components/patina_smbios/src/manager.rs

```diff
@@ -392,31 +348,158 @@ impl SmbiosManager {
             return Err(SmbiosError::OutOfResources);
         }
 
-        self.next_handle = candidate + 1;
+        *self.next_handle.borrow_mut() = candidate + 1;
         Ok(candidate)
     }
 
-    #[allow(dead_code)]
-    fn install_configuration_table(&self) -> Result<(), SmbiosError> {
-        // This would interact with UEFI Boot Services to install
-        // the SMBIOS 3.0+ table in the system configuration table
-        // Implementation depends on your UEFI framework
-
-        // For SMBIOS 3.0+
-        if let Some(_entry_point_64) = &self.entry_point_64 {
-            // Install with SMBIOS 3.0 GUID
+    /// Builds the SMBIOS table and installs it in the UEFI Configuration Table
+    ///
+    /// This function performs the following steps:
+    /// 1. Consolidates all SMBIOS records into a contiguous memory buffer
+    /// 2. Creates an SMBIOS 3.x Entry Point Structure with proper checksum
+    /// 3. Allocates ACPI Reclaim memory for both the table and entry point
+    /// 4. Installs the entry point via the UEFI Configuration Table
+    ///
+    /// # Arguments
+    ///
+    /// * `boot_services` - UEFI Boot Services for memory allocation and table installation
+    ///
+    /// # Returns
+    ///
+    /// Returns a tuple of `(table_address, entry_point_address)` containing the physical
+    /// addresses where the SMBIOS table data and entry point structure were allocated.
+    ///
+    /// # Errors
+    ///
+    /// * `SmbiosError::InvalidParameter` - No SMBIOS records have been added
+    /// * `SmbiosError::OutOfResources` - Failed to allocate memory or install the configuration table
+    ///
+    /// # Safety
+    ///
+    /// This function uses unsafe code for:
+    /// - Creating mutable slices to allocated memory
+    /// - Writing the entry point structure to allocated memory
+    /// - Calling the UEFI `install_configuration_table` interface
+    ///
+    /// All memory allocations use UEFI Boot Services and are properly tracked by the firmware.
+    pub fn install_configuration_table(
+        &self,
+        boot_services: &patina::boot_services::StandardBootServices,
+    ) -> Result<(PhysicalAddress, PhysicalAddress), SmbiosError> {
+        use patina::boot_services::allocation::{AllocType, MemoryType};
```

Comment:
We've typically been keeping use statements at the top of files, I'm not sure if we have a hard and fast style guideline there, but I think it makes the code easier to read.